Week3/homework/step2-2.js

```diff
@@ -2,19 +2,34 @@
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
-
   // Replace this comment and the next line with your code
   console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+  }
+  numbers.map(number => {
+    if (number % 3 === 0 && !(number % 5 === 0)) {
+      threeCallback(number);
+    } else if (number % 5 === 0 && !(number % 3 === 0)) {
+      fiveCallback(number);
+    } else if (number % 3 === 0 && number % 5 === 0) {
+      threeCallback(number);
+      fiveCallback(number);
+    }
+    return number;
+  });
 }
```

Comment:
You could simplify your `if-then-else` block by changing the order a bit:

```js
if (number % 3 === 0 && number % 5 === 0) {
  threeCallback(number);
  fiveCallback(number);
} else if (number % 3 === 0) {
  threeCallback(number);
} else if (number % 5 === 0) {
  fiveCallback(number);
}
```
If you test for the combined condition first, the second and the third conditions can be simpler.

However, still simpler and preferred is using two independent `if` statements, like this:

```js
numbers.forEach(number => {
  if (number % 3 === 0) {
    threeCallback(number);
  }
  if (number % 5 === 0) {
    fiveCallback(number);
  }
});
```

If you use `.forEach` instead of `.map` you can also get rid of the `return` statement without getting an ESLint error.